File: src/Email.java
import java.util.Scanner;

// Email class

public class Email {

    // define private variables
    private String firstName;
    private String lastName;
    private String department;

    private String email;
    private String password;
    private String altEmail;
    private int mailboxCapacity = 500;

    private int defaultPassLength = 10;
    private String companySuffix = "potato.com";

    // class constructor
    public Email(String firstName, String lastName) {
        System.out.println("~~~~~~~~~~~~~~~~~~~~");

        this.firstName = firstName;
        this.lastName = lastName;
        System.out.println("CREATING EMAIL FOR: " + firstName + " " + lastName);

        // ask for department by calling method
        this.department = setDepartment();

        // create email from above info
        this.email = firstName.toLowerCase() + "." + lastName.toLowerCase() + "@" + this.department + this.companySuffix;
        System.out.println("EMAIL HAS BEEN CREATED: " + this.email);

        // generate random password by calling method
        this.password = randomPassword(defaultPassLength);
        System.out.println("YOUR PASSWORD: " + this.password);

        System.out.println("~~~~~~~~~~~~~~~~~~~~");
    }
    
    // method to set department
    private String setDepartment() {
        Scanner scan = new Scanner(System.in);

        System.out.println("Departments:");
        System.out.println("1. HR");
        System.out.println("2. Marketing");
        System.out.println("3. Finance");
        System.out.println("4. IT");
        System.out.println("5. Operations");
        System.out.println("0. None");

        System.out.print("CHOOSE DEPARTMENT CODE: ");
        int code = scan.nextInt();
        scan.close();

        String department = "";

        switch (code) {
            case 1: department = "hr."; break;
            case 2: department = "marketing."; break;
            case 3: department = "finance."; break;
            case 4: department = "it."; break;
            case 5: department = "operations."; break;
            case 0: department = "";
        }

        return department;
    }

    // method to generate random password
    private String randomPassword(int length) {
        String passSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890!@£&?%#";
        
        char[] password = new char[length];

        for (int i = 0; i < length; i++) {
            int a = (int) (Math.random() * passSet.length());
            password[i] = passSet.charAt(a);
        }

        return String.valueOf(password);
    }

    // method to change password
    public void setPassword(String password) {
        this.password = password;
        System.out.println("Setting password...");
    }

    // method to change mailbox capacity
    public void setMailboxCapacity(int mailboxCapacity) {
        this.mailboxCapacity = mailboxCapacity;
        System.err.println("Setting mailbox capacity...");
    }

    // method to change alternate email
    public void setAltEmail(String altEmail) {
        this.altEmail = altEmail;
        System.err.println("Setting alternate email...");
    }

    // method to show information
    public void showInfo() {
        System.out.println("~~~~~~~~~~~~~~~~~~~~");
        System.out.println("Name: " + this.firstName + " " + this.lastName);
        System.out.println("Email: " + this.email);
        System.out.println("Password: " + this.password);
        System.out.println("Mailbox Capacity: " + this.mailboxCapacity);
        System.out.println("Alternate Email: " + this.altEmail);
        System.out.println("~~~~~~~~~~~~~~~~~~~~");
    }
    
}

File: bin/Email.py
from random import randint

# Create the same Email class as the one in Java


class Email:
    def __init__(self, first_name, last_name):
        print("~~~~~~~~~~~~~~~~~~~~")

        self.__first_name = first_name
        self.__last_name = last_name
        print(f"CREATING EMAIL FOR: {first_name} {last_name}")

        # ask for department by calling function
        self.__department = self.__set_department()

        # create email from above info
        self.__email = f"{first_name.lower()}.{last_name.lower()}@{self.__department}potato.com"
        print(f"EMAIL HAS BEEN CREATED: {self.__email}")

        # generate random password by calling function
        self.__password = self.__random_password()
        print(f"YOUR PASSWORD IS: {self.__password}")

        self.__mailbox_cap = 500
        self.__alt_email = None

        print("~~~~~~~~~~~~~~~~~~~~")

    # private function to set department
    def __set_department(self):

        print("Departments:")
        print("1. HR")
        print("2. Marketing")
        print("3. Finance")
        print("4. IT")
        print("5. Operations")
        print("0. None")

        code = int(input("CHOOSE DEPARTMENT CODE: "))

        if code == 1:
            return "hr."
        elif code == 2:
            return "marketing."
        elif code == 3:
            return "finance."
        elif code == 4:
            return "it."
        elif code == 5:
            return "operations."
        else:
            return ""

    # private function to generate random password
    def __random_password(self, length=10):
        characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890!@£&?%#"
        password = ["" for i in range(length)]

        for i in range(length):
            index = randint(0, len(characters) - 1)
            password[i] = characters[index]

        return "".join(password)

    # function to change password
    def set_password(self, password):
        print("Setting password...")
        self.__password = password

    # function to change mailbox capacity
    def set_mailbox_cap(self, capacity):
        print("Setting mailbox capacity...")
        self.__mailbox_cap = capacity

    # function to change alternate email
    def set_alternate_email(self, alt_email):
        print("Setting alternate email...")
        self.__alt_email = alt_email

    # function to show information
    def show_info(self):
        print("~~~~~~~~~~~~~~~~~~~~")
        print(f"Name: {self.__first_name} {self.__last_name}")
        print(f"Email: {self.__email}")
        print(f"Password: {self.__password}")
        print(f"Mailbox capacity: {self.__mailbox_cap}")
        print(f"Alternate email: {self.__alt_email}")
        print("~~~~~~~~~~~~~~~~~~~~")


email = Email("Frank", "Major")
email.set_password("C#4UgM3ntED")
email.set_mailbox_cap(350)
email.set_alternate_email("fsharpmajor@gmail.com")
email.show_info()
